assets/src/components/map.tsx

```diff
@@ -217,41 +223,108 @@ export const strokeOptions = ({ color }: Shape): object =>
         weight: 6,
       }
 
-const Shape = ({ shape }: { shape: Shape }) => {
-  const positions: LatLngExpression[] = shape.points.map((point) => [
-    point.lat,
-    point.lon,
-  ])
+const StopMarker = React.memo(
+  ({
+    stop,
+    iconSize,
+  }: {
+    stop: Stop
+    iconSize: StationIconSize
+  }): JSX.Element => {
+    if (stop.locationType === LocationType.Station) {
+      return <StationMarker station={stop} iconSize={iconSize} />
+    }
+
+    return (
+      <CircleMarker
+        className="m-vehicle-map__stop"
+        center={[stop.lat, stop.lon]}
+        radius={3}
+      >
+        <Popup className="m-vehicle-map__stop-tooltip">
+          {stop.name}
+          {inTestGroup(MAP_BETA_GROUP_NAME) && (
+            <StreetViewButton
+              latitude={stop.lat}
+              longitude={stop.lon}
+            ></StreetViewButton>
+          )}
+        </Popup>
+      </CircleMarker>
+    )
+  }
+)
+
+const StationMarker = ({
```

Comment:
**suggestion (non-blocking):** since `map.tsx` is getting kind of big, I wonder at what point we should split components like this out into other files. For instance we have separate `vehicleCard.tsx` and (soon) `stopCard.tsx` files, though it could be argued that those are larger components better suited for being separated out.

Reply:
Yeah I like that idea! Maybe as a starting point I'll move out all the markers to a single `mapMarkers` file.